Network/Converter/PacketConverter.cs

```diff
@@ -27,321 +28,680 @@
 //  You should have received a copy of the GNU Lesser General Public License
 //  along with this program.  If not, see <http://www.gnu.org/licenses/>.
 // ***********************************************************************
+
 #endregion Licence - LGPLv3
+
 using Network.Attributes;
+using Network.Extensions;
+using Network.Packets;
 using System;
+using System.Collections;
 using System.Collections.Generic;
 using System.IO;
-using System.Linq;
 using System.Reflection;
-using System.Collections;
-using Network.Extensions;
-using System.Runtime.Serialization;
 
 namespace Network.Converter
 {
     /// <summary>
-    /// Provides extension methods for packets to handle their read and write behaviors.
+    /// Implements <see cref="IPacketConverter"/>, and provides methods to serialise
+    /// and deserialise a packet to and from its binary form.
     /// </summary>
-    internal class PacketConverter : IPacketConverter
+    public class PacketConverter : IPacketConverter
```

Comment:
Keep the PacketConverter internal. Why public?

Reply:
People might want to use the packet converter? honestly IDK. will change back